### DISCUSSION TITLE
Viomi SE randomly loses wifi connection

### DISCUSSION THREAD

**Discussion by bedson87** [Hypfer/Valetudo#1735] [Q&A / Support]
Hi,
I have a problem with the Viomi SE robot. Latest valetudo firmware (lovely project!). Robot randomly loses connection to wifi. I noticed that quite often when he returns to the base after cleaning. I downloaded the logs via adb after disconnection (debug option). Wifi based on unifi ap with good range, SSID not hidden. It didn't happen on the original software, but here it happens several times a week or sometimes several times a day... very frustrating - I have to reboot the robot. 
Can I help you diagnose the problem?
Maybe it would be possible to make a script for robot that would reset wifi after losing connection every certain period of time?

[valetudo_debug.log](https://github.com/Hypfer/Valetudo/files/10900910/valetudo_debug.log)
[ip_a.log](https://github.com/Hypfer/Valetudo/files/10900920/ip_a.log)
![Zrzut ekranu z 2023-03-06 18-45-32](https://user-images.githubusercontent.com/37672068/223189886-e0cb2fa4-dc12-4c65-908c-fd406044cb36.png)
![Zrzut ekranu z 2023-03-06 18-46-50](https://user-images.githubusercontent.com/37672068/223189914-65ff85a5-e6e9-499d-8743-205b662c2d8f.png)

**Comment by APayerl** [comment]
I seem to have the same problem...
The interesting thing is that my robot lost contact around 8pm and still at 7am the next morning my router still has a dhcp lease for the device claiming its online... (lease time is 2h) So even though it does not communicate with home assistant via mqtt and the robots gui wbpage returns 404 the dhcp lease seems to refresh...

**Reply by BertilJ** [reply]
I've got the identical issue for a Dreamer robot with latest Valetudo on.

**Reply by APayerl** [reply]
My problem seem to have been resolved though after 1 restart. Then It has not happened again....

**Comment by rufik** [comment]
I'm using unify APs (AC-LR and U6-Lite, latest firmware) and my viomi connection looks really fine, no disconnections. At first I thought it's unable to roam between APs, but today I've checked controller logs deeper and I can see roaming events :)
But I experience some kind of "slowdowns" using UI or android app, after longer time of inactivity. I need to wait 20-30 seconds to get UI loaded, as if vacuum were sleeping.

**Comment by bedson87** [comment]
In another thread, I read to unblock ICMP to the gateway. After changing the firewall settings for the IoT VLAN, I haven't had any problems with the robot for a week. I don't know if it's that or just a coincidence, but for now it's okay
